### PR TITLE
allowed 'historical replay' of the noncustodial issue

### PR DESCRIPTION
**Pull request by andrewnguyen22**
- [ ] allows for historical replay of the noncust issue

### CODE REVIEW

**Comment by andrewnguyen22** [comment]
closes #1474

**Review by oten91 — COMMENTED** [review]
@andrewnguyen22 can you double check the "NonCustodial1RollbackHeight", as we are using <= while comparing to set "isDuringNonCustodialIssue". Everything seems in order but heavily dependent on this height.

An alternative strategy like using today's current height (or an arbitrary height we know the feature is disabled after the rollback) as a safe height to activate the new behaviour may be other option.

**Comment by iajrz** [comment]
@oten91 we're looking into the possibility of doing this fix differently from what we've done before, also without the `<=` vs `<` risk. Jessica created an issue for it in #1474; core idea is to not overload the network parameter and reap the benefits of that both in transparent behavior and easier/faster testing of the state machine permutations.

**Comment by andrewnguyen22** [comment]
What risk is there with `<` vs `<=` - either way, the flag was already rolled back so we are safe

**Comment by iajrz** [comment]
You are correct, I ran with Otto's idea without giving it much thought: in this particular case we're safe. I'm sure his observation is tied to past events around the codec change height and synchronization, so it's more an observation about safety in the general case rather than the specifics of this incident.

**Comment by oten91** [comment]
@iajrz Seem I'm missing context as the idea expressed on the issue is not super clear to me after reading your comment. 

Also I was under the impression that this was the related PR, and it seemed like a simple and good enough solution to tackle the sync from scratch scenario.

**Comment by iajrz** [comment]
Yes, this is the PR referred to from the issue. There seems to have been a miscommunication when the sync from scratch scenario was identified and solutions evaluated. We have alternatives to explore.

The core idea behind my proposal is this: as long as we add hardcoded heights to the executable we keep increasing the height a localnet needs to grow to in order to be able to fully test the scenarios.

The effect of this is that even if we were to have a localnet going through 100 heights per second a single scenario related to this would take over 10 minutes to test, an increase of over 30% of time with regards to the last height used to gate behavior.

There is another argument to be made: using the same flag for more than one functionality unfavorably shapes testing parameter combinations.

I think we can discuss this when you get back. Get better.

File: codec/codec.go
Comment (Olshansk) [inline codec/codec.go:42]: What is the purpose of the `1` in `NonCustodial1RollbackHeight`?

File: x/nodes/keeper/reward.go
Comment (Olshansk) [inline x/nodes/keeper/reward.go:32]: Not actionable: good learning lesson for myself on the importance of blockchain history & immutability.

**Review by Olshansk — APPROVED** [review]
Code LGTM. Left one comment but would also want an approval from @oten91 before merge.

**Review by oten91 — APPROVED** [review]

**Comment by oten91** [comment]
For context , documentation and answering my original ask of double checking the height, I double checked it and height 69583 should work without problems:

This is the rollback tx https://pokt.watch/tx/BD2AB44335EE3355EDECDAA10613602D0CEAD2A5D57B518E5ECA1F2678C022CC .

**Pull request merged**
